src/components/ConnectMetamask.jsx

```diff
@@ -21,11 +23,17 @@ init({
 export default function ConnectMetamask() {
 
     const [{ wallet, connecting }, connect, disconnect] = useConnectWallet();
+    const dispatch = useDispatch();
 
     useEffect(() => {
         if (!window.ethereum) {
             alert("install MetaMask");
         }
+        if (wallet) {
+            console.log(wallet.accounts[0].address)
+        } else {
+            dispatch(setAddress(null));           

```

Comment:
The only place we set the address to null is on logout or on network/account change (we don't support network changes)